### ISSUE TITLE
AssertionError relating to bond directions around double bond when converting to RDKit molecule

### ISSUE THREAD

**Issue by rosemary-cresset** [openforcefield/openff-toolkit#1190]
**Describe the bug**
When running the following code snippet, I hit an error in the conversion from OpenFF molecule to RDKit molecule. RDKit handles the SMILES string without complaint. 

**To Reproduce**
```
from openff.toolkit.topology import Molecule

smiles = '[H]/[C]([C]1=[C]([C]([H])([H])[H])[C]([H])([H])[C]([H])([H])[C]([H])([H])[C@@]1([C]([H])([H])[H])[C]([H])([H])[H])=[C]([H])\[C](=[C]([H])\[C]([H])=[C]([H])\[C](=[C](/[H])[C]([H])([H])[H])[C]([H])([H])[H])[C]([H])([H])[H]'

openff_mol = Molecule.from_smiles(smiles)
openff_mol.to_rdkit()
```

**Output**
```
Warning: Unable to load toolkit 'OpenEye Toolkit'. The Open Force Field Toolkit does not require the OpenEye Toolkits, and can use RDKit/AmberTools instead. However, if you have a valid license for the OpenEye Toolkits, consider installing them for faster performance and additional file format support: https://docs.eyesopen.com/toolkits/python/quickstart-python/linuxosx.html OpenEye offers free Toolkit licenses for academics: https://www.eyesopen.com/academic-licensing
[15:30:50] Conflicting single bond directions around double bond at index 25.
[15:30:50]   BondStereo set to STEREONONE and single bond directions set to NONE.
Traceback (most recent call last):
  File "/tmp/test/test.py", line 11, in <module>
    openff_mol.to_rdkit()
  File "/data/rosemary/anaconda3/envs/openfftoolkit/lib/python3.9/site-packages/openff/toolkit/utils/base_wrapper.py", line 51, in wrapped_function
    value = func(*args, **kwargs)
  File "/data/rosemary/anaconda3/envs/openfftoolkit/lib/python3.9/site-packages/openff/toolkit/topology/molecule.py", line 5147, in to_rdkit
    return toolkit.to_rdkit(self, aromaticity_model=aromaticity_model)
  File "/data/rosemary/anaconda3/envs/openfftoolkit/lib/python3.9/site-packages/openff/toolkit/utils/rdkit_wrapper.py", line 1794, in to_rdkit
    cls._assign_rdmol_bonds_stereo(molecule, rdmol)
  File "/data/rosemary/anaconda3/envs/openfftoolkit/lib/python3.9/site-packages/openff/toolkit/utils/rdkit_wrapper.py", line 2406, in _assign_rdmol_bonds_stereo
    assert stereo_rdbond.GetStereo() == desired_rdk_stereo_code
AssertionError
```

**Computing environment (please complete the following information):**
 - Operating system: Linux
 - Output of running `conda list`:
```
# packages in environment at /data/rosemary/anaconda3/envs/openfftoolkit:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
amberlite                 16.0                     pypi_0    pypi
ambertools                21.11            py39hc630cb1_0    conda-forge
amberutils                21.0                     pypi_0    pypi
arpack                    3.7.0                hdefa2d7_2    conda-forge
astunparse                1.6.3              pyhd8ed1ab_0    conda-forge
blosc                     1.21.0               h9c3ff4c_0    conda-forge
boost                     1.74.0           py39h5472131_5    conda-forge
boost-cpp                 1.74.0               h6cacc03_7    conda-forge
brotli                    1.0.9                h7f98852_6    conda-forge
brotli-bin                1.0.9                h7f98852_6    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.18.1               h7f98852_0    conda-forge
ca-certificates           2021.10.8            ha878542_0    conda-forge
cairo                     1.16.0            ha00ac49_1009    conda-forge
certifi                   2021.10.8        py39hf3d152e_1    conda-forge
cudatoolkit               11.6.0              habf752d_10    conda-forge
curl                      7.81.0               h2574ce0_0    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cython                    0.29.27          py39he80948d_0    conda-forge
fftw                      3.3.10          nompi_h77c792f_102    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 hab24e00_0    conda-forge
fontconfig                2.13.96              ha180cfb_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.29.1           py39h3811e60_0    conda-forge
freetype                  2.10.4               h0708190_1    conda-forge
fribidi                   1.0.10               h36c2ea0_0    conda-forge
gettext                   0.19.8.1          h73d1719_1008    conda-forge
giflib                    5.2.1                h36c2ea0_2    conda-forge
greenlet                  1.1.2            py39he80948d_1    conda-forge
hdf4                      4.2.15               h10796ff_3    conda-forge
hdf5                      1.12.1          nompi_h2750804_103    conda-forge
icu                       69.1                 h9c3ff4c_0    conda-forge
importlib-metadata        4.10.1           py39hf3d152e_0    conda-forge
jbig                      2.1               h7f98852_2003    conda-forge
jpeg                      9e                   h7f98852_0    conda-forge
kiwisolver                1.3.2            py39h1a9c180_1    conda-forge
krb5                      1.19.2               hcc1bbae_3    conda-forge
lcms2                     2.12                 hddcbb42_0    conda-forge
ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
lerc                      3.0                  h9c3ff4c_0    conda-forge
libblas                   3.9.0           13_linux64_openblas    conda-forge
libbrotlicommon           1.0.9                h7f98852_6    conda-forge
libbrotlidec              1.0.9                h7f98852_6    conda-forge
libbrotlienc              1.0.9                h7f98852_6    conda-forge
libcblas                  3.9.0           13_linux64_openblas    conda-forge
libcurl                   7.81.0               h2574ce0_0    conda-forge
libdeflate                1.8                  h7f98852_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 11.2.0              h1d223b6_12    conda-forge
libgfortran-ng            11.2.0              h69a702a_12    conda-forge
libgfortran5              11.2.0              h5c6108e_12    conda-forge
libglib                   2.70.2               h174f98d_2    conda-forge
libgomp                   11.2.0              h1d223b6_12    conda-forge
libiconv                  1.16                 h516909a_0    conda-forge
libimagequant             2.17.0               h7f98852_1    conda-forge
liblapack                 3.9.0           13_linux64_openblas    conda-forge
libnetcdf                 4.8.1           nompi_hb3fd0d9_101    conda-forge
libnghttp2                1.46.0               h812cca2_0    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libopenblas               0.3.18          pthreads_h8fe5266_0    conda-forge
libpng                    1.6.37               h21135ba_2    conda-forge
libssh2                   1.10.0               ha56f1ee_2    conda-forge
libstdcxx-ng              11.2.0              he4da1e4_12    conda-forge
libtiff                   4.3.0                h6f004c6_2    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libwebp                   1.2.2                h3452ae3_0    conda-forge
libwebp-base              1.2.2                h7f98852_1    conda-forge
libxcb                    1.13              h7f98852_1004    conda-forge
libxml2                   2.9.12               h885dcf4_1    conda-forge
libzip                    1.8.0                h4de3113_1    conda-forge
libzlib                   1.2.11            h36c2ea0_1013    conda-forge
lz4-c                     1.9.3                h9c3ff4c_1    conda-forge
lzo                       2.10              h516909a_1000    conda-forge
matplotlib-base           3.5.1            py39h2fa2bec_0    conda-forge
mdtraj                    1.9.7            py39h138c130_1    conda-forge
mmpbsa-py                 16.0                     pypi_0    pypi
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
ncurses                   6.3                  h9c3ff4c_0    conda-forge
netcdf-fortran            4.5.4           nompi_h2b6e579_100    conda-forge
networkx                  2.6.3              pyhd8ed1ab_1    conda-forge
nomkl                     1.0                  h5ca1d4c_0    conda-forge
numexpr                   2.8.0           py39hbd72853_101    conda-forge
numpy                     1.22.2           py39h91f2184_0    conda-forge
ocl-icd                   2.3.1                h7f98852_0    conda-forge
ocl-icd-system            1.0.0                         1    conda-forge
openff-forcefields        2.0.0              pyh6c4a22f_0    conda-forge
openff-toolkit            0.10.2             pyhd8ed1ab_0    conda-forge
openff-toolkit-base       0.10.2             pyhd8ed1ab_0    conda-forge
openjpeg                  2.4.0                hb52868f_1    conda-forge
openmm                    7.7.0            py39h792354b_0    conda-forge
openssl                   1.1.1l               h7f98852_0    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
packmol                   20.010               h86c2bf4_0    conda-forge
packmol-memgen            1.2.1rc0                 pypi_0    pypi
pandas                    1.4.0            py39hde0f152_0    conda-forge
parmed                    3.4.3            py39he80948d_1    conda-forge
pcre                      8.45                 h9c3ff4c_0    conda-forge
pdb4amber                 20.1                     pypi_0    pypi
perl                      5.32.1          1_h7f98852_perl5    conda-forge
pillow                    9.0.1            py39he69867a_0    conda-forge
pip                       22.0.3             pyhd8ed1ab_0    conda-forge
pixman                    0.40.0               h36c2ea0_0    conda-forge
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
pycairo                   1.20.1           py39hedcb9fc_1    conda-forge
pyparsing                 3.0.7              pyhd8ed1ab_0    conda-forge
pytables                  3.7.0            py39h2669a42_0    conda-forge
python                    3.9.10          h85951f9_2_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.9                      2_cp39    conda-forge
pytraj                    2.0.6                    pypi_0    pypi
pytz                      2021.3             pyhd8ed1ab_0    conda-forge
rdkit                     2021.09.4        py39hccf6a74_0    conda-forge
readline                  8.1                  h46c0cb4_0    conda-forge
reportlab                 3.5.68           py39he59360d_1    conda-forge
sander                    16.0                     pypi_0    pypi
scipy                     1.8.0            py39hee8e79c_0    conda-forge
setuptools                60.8.1           py39hf3d152e_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
smirnoff99frosst          1.1.0              pyh44b312d_0    conda-forge
snappy                    1.1.8                he1b5a44_3    conda-forge
sqlalchemy                1.4.31           py39h3811e60_0    conda-forge
sqlite                    3.37.0               h9cd32fc_0    conda-forge
tk                        8.6.11               h27826a3_1    conda-forge
tzdata                    2021e                he74cb21_0    conda-forge
unicodedata2              14.0.0           py39h3811e60_0    conda-forge
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
xmltodict                 0.12.0                     py_0    conda-forge
xorg-kbproto              1.0.7             h7f98852_1002    conda-forge
xorg-libice               1.0.10               h7f98852_0    conda-forge
xorg-libsm                1.2.3             hd9c2040_1000    conda-forge
xorg-libx11               1.7.2                h7f98852_0    conda-forge
xorg-libxau               1.0.9                h7f98852_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xorg-libxext              1.3.4                h7f98852_1    conda-forge
xorg-libxrender           0.9.10            h7f98852_1003    conda-forge
xorg-libxt                1.2.1                h7f98852_2    conda-forge
xorg-renderproto          0.11.1            h7f98852_1002    conda-forge
xorg-xextproto            7.3.0             h7f98852_1002    conda-forge
xorg-xproto               7.0.31            h7f98852_1007    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
zipp                      3.7.0              pyhd8ed1ab_1    conda-forge
zlib                      1.2.11            h36c2ea0_1013    conda-forge
zstd                      1.5.2                ha95c52a_0    conda-forge
```

**Comment by j-wags** [comment]
Hi @rosemary-cresset,

Thanks for the great bug report - I was able to quickly reproduce this error with the current codebase. 

I believe that the root cause for this issue is the same as #1181. Thankfully, we are working on a fix in #1184, and your excellent reproducing example passes when I run it using the code in that branch. So you should expect this to be fixed in our next release :-)

I'll leave this issue open, and link it in #1184 so it gets closed when we merge that PR.

**Comment by rosemary-cresset** [comment]
Thanks, that's great. :-)
